tests/cupy_tests/core_tests/test_ndarray_reduction.py

```diff
@@ -228,6 +232,25 @@ def test_cub_min(self, xp, dtype, axis):
             a = xp.ascontiguousarray(a)
         elif self.order in ('f', 'F'):
             a = xp.asfortranarray(a)
+
+        if xp is numpy:
+            return a.min(axis=axis)
+
+        # xp is cupy, first ensure we really use CUB
+        full_reduction = 'cupy.core._routines_statistics.cub.device_reduce'
+        full_raise = NotImplementedError('gotcha_full')
+        segmented_reduction = ('cupy.core._routines_statistics.cub.'
+                               'device_segmented_reduce')
+        segmented_raise = NotImplementedError('gotcha_segment')
+        with mock.patch(full_reduction, side_effect=full_raise), \
+                mock.patch(segmented_reduction, side_effect=segmented_raise), \
+                pytest.raises(NotImplementedError) as e:
+            a.min(axis=axis)
```

Comment:
I am aware that the style guide is against using backslashes, but this probably needs to be the only exception. Multiple context managers aren't properly handled in Python or all linters, including black (currently managed by Python core devs), so there's just no way to make them formatted pretty. See this related issue from the black repo: psf/black#412.

Reply:
Very clear explanation: https://github.com/psf/black/issues/664#issuecomment-594771153.

Reply:
Found a silly way to work around this...